### PR TITLE
[ci] Run Android unit tests even if spotless check fails

### PR DESCRIPTION
**Pull request by vonovak**
# Why

- the `Run native Android unit tests` step only runs if spotless passes. I'd like it to run even if spotless fails and see the result
- many jobs have `cancel-in-progress: true`. I believe it's beneficial to leave them running and see their output as I'm pushing more changes, rather then killing them eagerly. 2 of 3 where I switched to `false` are running on EAS in case concurrency is a problem

# How

updated workflows

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
> since our ci resource is limited, i'm wondering if that's necessary? changing these may save personal time, but it will get all the other persons to get blocked:
> 

It isn't really necessary but I sometimes thought it'd be nice so I came up with this diff. If there are concurrency issues then let's not merge. Personally, I haven't noticed long waiting times for CI to start running.

As for continuing the workflow even after `spotlessCheck`fails - the rationale is that I want to get as much feedback as possible early. If code is badly formatted, and a fault will cause unit tests to fail, first CI run tells me `spotlessCheck` is failing. I need to run again to learn that unit tests are failing too. If I can get both pieces of information in a single run, it's a net saving. Of course the question is how often that's gonna happen...

**Comment by Kudo** [comment]
> It isn't really necessary but I sometimes thought it'd be nice so I came up with this diff. If there are concurrency issues then let's not merge. Personally, I haven't noticed long waiting times for CI to start running.

it happens. it could be the reason why we didn't include `package/**/*` changing scope to trigger eas build:
https://github.com/expo/expo/blob/f842a94a9ec5225893244c7c9b2bb0666c29e6c3/.github/workflows/client-ios-eas.yml#L7-L14
i would also like to merge all updates e2e to a single custom eas build at some point 🙈

File: .github/workflows/android-unit-tests.yml [hunk 1]
```diff
@@ -65,6 +65,7 @@ jobs:
         working-directory: apps/expo-go/android
         run: ./gradlew spotlessCheck || { echo '::error Spotless lint failed. Run `./gradlew spotlessApply` from `apps/expo-go/android` to automatically fix formatting.' && exit 1; }
       - name: 🎸 Run native Android unit tests
+        if: always()
```
Comment (tsapeta) [inline .github/workflows/android-unit-tests.yml:68]: If the spotless check fails, but native unit tests pass, will the job finish as failed or succeeded?

**Review by tsapeta — COMMENTED** [review]
I agree with Kudo. I know that sometimes CI could be more helpful, but our resources are limited and we need to care about them too. 

In theory, you as an author of the PR should run some of these checks locally before opening a PR and requesting for review. CI is only like a guard and an information for reviewers (and someone who merges) that the code you want to be merged works and doesn't break other's work. Let's not overuse CI and EAS resources for something that can easily be run locally. I use `[skip ci]` tags a lot when I push something to my PRs but don't need to trigger the CI – especially when I apply bot's suggestions to add links in the changelog.

File: .github/workflows/android-unit-tests.yml (hunk 1 above)
Comment (vonovak) [reply .github/workflows/android-unit-tests.yml:68]: failed

Comment (tsapeta) [reply .github/workflows/android-unit-tests.yml:68]: So I'm ok with this specific change 👍

Comment (Kudo) [reply .github/workflows/android-unit-tests.yml:68]: also fine for this change given ubuntu runners have more availability.

**Review by Kudo — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**